src/expression.js

```diff
@@ -706,7 +706,7 @@ pp.parseFunctionBody = function(node, isArrowFunction) {
     // flag (restore them to their old value afterwards).
     let oldLabels = this.labels
     let oldVarDeclaredNames = this.varDeclaredNames
-    this.varDeclaredNames = new Set()
+    this.varDeclaredNames = {}
```

Comment:
I had avoided using `Object.create` because it's an ES5 feature. Are some ES5 builtins supported?